### PR TITLE
Log errors when logging is enabled

### PR DESCRIPTION
**Pull request by iamajvillalobos**
- Enable logging for possible json errors
- Strip special characters when converting string to json

### CODE REVIEW

**Comment by ramontayag** [comment]
@iamajvillalobos were you able to figure out why it's failing?

File: lib/ezapi_client/services/gen_data.rb [hunk 1]
```diff
@@ -44,9 +44,16 @@ def default_command
...
       message.each_with_object({}) do |(key, value), hash|
-        hash[key.to_s.camelcase(:lower)] = value
+        hash[key.to_s.camelcase(:lower)] = strip_special_characters(value)
       end.to_json
     end
 
+    def strip_special_characters(value)
+      return value if !value.is_a?(String)
+
+      rule = /[^0-9A-Za-z ]/
```
Comment (ramontayag) [inline lib/ezapi_client/services/gen_data.rb]: This looks like it should be a frozen constant :+1:

We should allow `-` though.

```ruby
/[^\w-\s]/
```

☝️  "not a digit or letter, not a hyphen, not a space"

**Review by ramontayag — CHANGES_REQUESTED** [review]

File: lib/ezapi_client/services/exec_command.rb [hunk 2]
```diff
@@ -4,18 +4,23 @@ class ExecCommand
     include Virtus.model
     attribute :command, String
 
-    def self.call(command)
-      self.new(command: command).()
+    def self.call(command, logger = nil)
+      if logger
+        self.new(command: command).(logger)
+      else
+        self.new(command: command).()
```
Comment (ramontayag) [inline lib/ezapi_client/services/exec_command.rb]: Oh you can always pass logger. No need for this since you already have the `if` in line 23

**Review by ramontayag — CHANGES_REQUESTED** [review]

File: lib/ezapi_client/services/exec_command.rb (hunk 2 above)
Comment (iamajvillalobos) [reply lib/ezapi_client/services/exec_command.rb]: ah yes! thanks for the catch, I added the if on line 23 on a later time.

**Comment by iamajvillalobos** [comment]
@ramontayag updated the PR with your suggestion on the logger

**Review by ramontayag — APPROVED** [review]
💯

**Pull request merged**